welcome path gotcha tip (#927)
jamulussoftware/jamulussoftware.github.io: wiki/en/Running-a-Server.md

##### `-w or --welcomemessage`
A "welcome message" to display in the Client chat window on connect. Can be given as a string or filename, and can contain HTML.

> When a path is used, the full path must be accessible by the `jamulus` Linux user. If not accessible, the literal path (rather than its contents) will appear.

Reviewer: > the `jamulus` Linux user

This should refer to "the user running Jamulus", because it applies regardless of operating system and who runs it.

I'd have to check the source but the file should only need to be readable (i.e. if as the user running Jamulus, you can read the file, then Jamulus running as that user can read the file and will use it).

Author: I know there can be a "we rely on operators" approach around here, but I prefer assuming the user is completely ignorant about the nuances of platform access controls. Comprehending which user is running Jamulus is a reasonably heavy mental lift to me. The `jamulus` user appears in the `systemctl --edit` configuration file. I suspect this user gets created by systemd if it does not exist already, and it probably does not. Which means by default, there is no `/home/jamulus`, and even if there was, a security feature we use would prevent this process from reading from /home/* anyway. So specifying this user is misleading, because they don't have that interactive user file path, and we couldn't use it if they did.

Creating `/etc/jamulus` was suggested, which probably entails a **mkdir** and then a **chown**. If we provide a little three-line pasteable bash script for that, maybe that's not a bad solution.

> On Linux or Mac, enter:
> 
> ```
> mkdir /etc/jamulus
> chown jamulus /etc/jamulus
> echo Welcome > /etc/jamulus/welc.html
> ```
> 
> Example: ... -w /etc/jamulus/welc.html

Reviewer 2: What about "the user running the Server (e.g. `jamulus` on the default Linux install)"?

Author: Do you feel that's better than the draft I wrote in the previous comment?

Problem 1: /home/jamulus doesn't exist. So... what user? On Day 1, Linus told us some user accounts don't have interactive /home/username space. But I slept through Day 1.
Problem 2: Even if the user just creates an interactive /home/jamulus, they can't use it. Hope they don't try, because
Problem 3: There are no error explanations.

So I like recommending /etc/jamulus/welc.html on Linux and Mac, if that would work. It's not really clear where such a file can and should go.

Reviewer 2: I mean there's probably no "jamulus" user on macOS or differently set up linux systems. 
Yes, the /etc/jamulus/ path would be ok if you think of it as a config file. (I think I store it under /var/jamulus/ for my server - but I don't remember).

Reviewer 2: If you install the deb file a user called jamulus gets created. See the postinst file. (At the moment it's here: https://github.com/jamulussoftware/jamulus/blob/main/linux/debian/jamulus-headless.postinst)

Reviewer: You don't need to run on a Debian-based Linux OS system to run Jamulus as a server.  Any documentation needs to cover all cases or be generic enough to cover all cases.

Author: Covering all cases isn't needed. Generic plus major platforms is enough. Generic without major platforms results in friction for most readers. With 20 years writing docs, I have 20 years of this debate with lead engineers. Be skeptical of omissions and support explicit guidance for the biggest platforms.

> When a path is used, the full path must be accessible by the `jamulus` Linux user. If not accessible, the literal path (rather than its contents) will appear.

Author: ```suggestion
When a path is used, the file must be accessible by the user account running Jamulus. (On Linux and Mac, user `jamulus` is used by default.) If not accessible, the literal path (rather than its contents) will appear. 
```

> When a path is used, the file must be accessible by the user account running Jamulus. (On Linux and Mac, user `jamulus` is used by default.) If not accessible, the literal path (rather than its contents) will appear. 

Author: ```suggestion
When a path is used, the file must be accessible by the user account running Jamulus. (On most Linux installations, user `jamulus` is used by default.) If not accessible, the literal path (rather than its contents) will appear. 
```
